spec/ddtrace/contrib/concurrent_ruby/integration_test_spec.rb

```diff
@@ -1,45 +1,27 @@
-require 'concurrent/future'
+require 'concurrent-ruby' # concurrent-ruby is not modular
 
 require 'ddtrace/contrib/support/spec_helper'
 require 'ddtrace'
 require 'spec/support/thread_helpers'
 
 RSpec.describe 'ConcurrentRuby integration tests' do
-  # DEV We save an unmodified copy of Concurrent::Future.
-  let!(:unmodified_future) { ::Concurrent::Future.dup }
   let(:configuration_options) { {} }
   let(:outer_span) { spans.find { |s| s.name == 'outer_span' } }
   let(:inner_span) { spans.find { |s| s.name == 'inner_span' } }
 
-  before(:context) do
-    # Execute an async future to force the eager creation of internal
-    # global threads that are never closed.
-    #
-    # This allows us to separate internal concurrent-ruby threads
-    # from ddtrace threads for leak detection.
-    ThreadHelpers.with_leaky_thread_creation(:concurrent_ruby) do
-      Concurrent::Future.execute {}.value
-    end
-  end
+  # DEV We save an unmodified copy of Concurrent::Future.
+  let!(:unmodified_promises) { ::Concurrent::Promises.dup }
+  let!(:unmodified_future) { ::Concurrent::Future.dup }
 
   # DEV We then restore Concurrent::Future, a dangerous game.
   after do
+    ::Concurrent.send(:remove_const, :Promises)
+    ::Concurrent.const_set('Promises', unmodified_promises)
     ::Concurrent.send(:remove_const, :Future)
     ::Concurrent.const_set('Future', unmodified_future)
     remove_patch!(:concurrent_ruby)
   end
```

Comment:
I wonder if we could use RSpec's constant stubbing for this purpose, rather than doing our own homegrown constant stubbing implementation.

Reply:
> My understanding is that we want to test on the real constant (integration testing) but we also want to be able to restore the original inheritance chain (Ideally we would just remove the module from the chain, but that's not possible in Ruby. I had on occasion resorted to inserting dynamically created modules, and empty them afterwards).

To be honest, I think the could just dependency inject the modules to modify and have the production ones as defaults, but that's usually not what has been done in the gem historically (I also think we shouldn't have as many module functions) so I didn't suggest going that far :)

> Are you suggesting we could do something like this to achieve a similar effect?

Yeah, I was thinking that they should be equivalent. I can't think of any cases where it wouldn't be -- but it may just be an unknown-unknown.

Reply:
This `after` block is pure dark magic.
It works, but no one knows if it will continue to after the next change.

If you can find a better way to reset patching, please go ahead and make that change.

Alternatively, this is the only integration that does this: maybe such strict testing is not truly required?

Either way, you might also just leave it as is if you can't find a good alternative.

Reply:
Moved to `stub_const` which seems to work well, and is more readable.

I do have some facilities to handle this in the upcoming Graft instrumentation, but further change here feels like being way out of scope for this PR. The other ways I can come up with feel even hackier.

As for testing this, pragmatically I definitely found these tests useful to check that this PR works.